src/App.jsx

```diff
@@ -11,10 +14,30 @@ function App() {
   const [todoList, setTodoList] = useState(onLoadList || []);
   const [isLoading, setIsLoading] = useState(true);
 
+  const [filteredTodos, setFilteredTodos] = useState(todoList);
```

Comment:
I don't recommend setting one state variable (`todoList` in this case) as a default state for another state variable. It's redundant and could be confusing. Instead, you could reuse your existing `todoList` state variable like so:

```js
const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem("savedTodoList")) || []);
const [isLoading, setIsLoading] = useState(true);
const [searchTerm, setSearchTerm] = useState("");

const filteredTodos = useMemo(() => {
  if (searchTerm === "") {
    return todoList;
  }
  return todoList.filter((todo) =>
    todo.title.toLowerCase().includes(searchTerm.toLowerCase())
  );
}, [todoList, searchTerm]);

const handleSearch = (term) => {
  setSearchTerm(term);
};
```

Instead we can store the searchTerm as a state variable and react to its changes. Like useEffect, [useMemo](https://react.dev/reference/react/useMemo) takes a dependency array where you can specify reactive values. This means that `filteredTodos` will be recalculated when a new `searchTerm` is set.